pymodbus/server/reactive/main.py

```diff
@@ -92,6 +102,85 @@
 """
 
 
+class ReactiveModbusSlaveContext(ModbusSlaveContext):
+    """Reactive Modbus slave context"""
+
+    def __init__(
+        self,
+        discrete_inputs: BaseModbusDataBlock = None,
+        coils: BaseModbusDataBlock = None,
+        input_registers: BaseModbusDataBlock = None,
+        holding_registers: BaseModbusDataBlock = None,
+        zero_mode: bool = False,
+        randomize: int = 0,
+        **kwargs,
+    ):
+        """Reactive Modbus slave context supporting simulating data.
+
+        :param discrete_inputs: Discrete input data block
+        :param coils: Coils data block
+        :param input_registers: Input registers data block
+        :param holding_registers: Holding registers data block
+        :param zero_mode: Enable zero mode for data blocks
+        :param randomize: Randomize reads every <n> reads for DI and IR,
+                          default is disabled (0)
+        :param min_binary_value: Minimum value for coils and discrete inputs
+        :param max_binary_value: Max value for discrete inputs
+        :param min_register_value: Minimum value for input registers
+        :param max_register_value: Max value for input registers
+
+        """
+        super().__init__(
+            di=discrete_inputs,
+            co=coils,
+            ir=input_registers,
+            hr=holding_registers,
+            zero_mode=zero_mode,
+        )
+        min_binary_value = kwargs.get("min_binary_value", 0)
+        max_binary_value = kwargs.get("max_binary_value", 1)
+        min_register_value = kwargs.get("min_register_value", 0)
+        max_register_value = kwargs.get("max_register_value", 65535)
+        self._randomize = randomize
+        self._lock = threading.Lock()
+        self._read_counter = {"d": 0, "i": 0}
+        self._min_binary_value = min_binary_value
+        self._max_binary_value = max_binary_value
+        self._min_register_value = min_register_value
+        self._max_register_value = max_register_value
+
+    def getValues(self, fc_as_hex, address, count=1):
```

Comment:
should this functionality work for single bit object types? I was thinking about coils.
Not sure if is really much value, but for my own tests that would be nice, although the complexity of randomise booleans is much less.

Anyhow, just wondering about your thoughts about those objects.

Reply:
Holding registers and coils can be overridden and hence they are excluded from simulation. Bit fields should be straight forward (0 or 1)

Reply:
Right that makes sense